### DISCUSSION TITLE
Working Much Better - Minor Issues

### DISCUSSION THREAD

**Discussion by Martin2112** [nschlia/ffmpegfs#132] [Ideas]
Hi,

Things have been going well since the fixes. I'm left with a few minor points, some of them not really ffmpegfs related.

1.) Cue sheets containing characters not valid in windows filenames such as a colon. This works fine in Linux but causes problems if exporting ffmpegfs filesystems with Samba. Can be worked around by defining character mappings in smb.conf. Not sure it's worth trying to filter them out as there are a few and it only affects Windows.

2.) Exact Audio Copy has stored the cue sheet in the files as a text tag (CUESHEET). This appears in all the track files and seems to confuse players like VLC that the track name is always track one from the cue sheet. This might be a VLC bug though as the media information shows the correct track number. Maybe this tag could be filtered out during the transcode?

3.) The input large FLAC files with complete albums appear in the output tree. This causes players like Volumio to create additional entries for the album as well as the tracks. This is a minor irritation but can be confusing when one of the album tracks has the same name as the album itself. I wondered if it would be possible to optionally hide these?

**Comment by nschlia** [comment]
> Things have been going well since the fixes. I'm left with a few minor points, some of them not really ffmpegfs related.

Great. Thanks for you efforts, I'll try to work them out ASAP, but currently I am a bit busy. Also working on the subtitle feature. So this may take a bit.

> 1.) Cue sheets containing characters not valid in windows filenames such as a colon. This works fine in Linux but causes problems if exporting ffmpegfs filesystems with Samba. Can be worked around by defining character mappings in smb.conf. Not sure it's worth trying to filter them out as there are a few and it only affects Windows.

I guess I should transform file names generally to a Windoze compatible format. Done that once before for another project, guess I can nick my old code that is working for 20 years or so now without customer complaints.
 
> 2.) Exact Audio Copy has stored the cue sheet in the files as a text tag (CUESHEET). This appears in all the track files and seems to confuse players like VLC that the track name is always track one from the cue sheet. This might be a VLC bug though as the media information shows the correct track number. Maybe this tag could be filtered out during the transcode?

You mean that the CUESHEET tag is added to all virtual tracks causes havoc? Should be no problem to just not do so. FFmpegfs brainlessly copies all tags over, it could easily drop the CUESHEET that makes no sense in them anyways. 

> 3.) The input large FLAC files with complete albums appear in the output tree. This causes players like Volumio to create additional entries for the album as well as the tracks. This is a minor irritation but can be confusing when one of the album tracks has the same name as the album itself. I wondered if it would be possible to optionally hide these?

Thought it'd be nice to have the large file as well. If this is an issue it should be hidden. I could add a command line option to make it visible but hide it by default.

**Reply by Martin2112** [reply]
OK thanks, 

For 2.) The CUESHEET tag seems to confuse some players, I think some don't read it or work OK. It's not a major breakage but it would be a nice to have.

On 3.) An option sounds best. as some people might want the big files. Again, it's not a big issue, just a nice to have.

**Comment by Martin2112** [comment]
Oh, another possible idea - some randomization in cache expiry. If you have a player that scans for music you end up with everything having almost the same expiry time.

If you could set e.g to expire in 3 months plus or minus one month then it would spread things out more.

**Reply by jmckee0** [reply]
Martin, I see that you forked the project.  Did you figure out how to omit source files for cuesheet splitting?   I can't imagine it would take more than commenting or altering a few lines.    Unfortunately I am an amateur coder and c++ is a bit opaque to me.

**Comment by Martin2112** [comment]
Hi,

I only forked it to create a patch. Several options were added by nschlia
including --hide_extensions and --include_extensions. I think it should be
able to do what you want without code changes..

Martin

**Reply by jmckee0** [reply]
It's not immediately apparent how to use those options to selectively hide files with the same extension but I will tinker with it, thanks. 
You mentioned this in your first post (point #3) so I assumed it was not possible with current options..

**Reply by jmckee0** [reply]
figured this out:
fuseops.cc
...
                            if (!(flags & VIRTUALFLAG_HIDDEN))                                                                                       
                            {
                                int yogurt = check_cuesheet(origfile, buf, filler);
                                if (yogurt < 0)
                                {
                                    throw EINVAL;                                                                                                    
                                }                                                                                                                    
                                if (yogurt > 0)                                                                                                        
                                {
                                    continue;                                                                                                        
                                }                                                                                                                    
                            }                                                                                                                        
                ...
Working really well with Navidrome now.  However it crashes when scanned by Jellyfin.  I will post some logs of that in the issues forum when I get a chance.

**Reply by nschlia** [reply]
Interesting, a crash with one client, none with another?

Can you provide core dumps?
